### DISCUSSION TITLE
Mass mailing

### DISCUSSION THREAD

**Discussion by murraycollingwood** [PHPMailer/PHPMailer#2917] [General]
Summary: I want to send personalised emails to perhaps 500 different email addresses.  Previously I have done this by sending each individual email, but now I want to be more efficient, opening a single SMTP session and sending each email with just an RSET between each.  I read the documentation but couldn't find the answer - can you answer this question?

Previously I would setup the email content and headers, and then call $mail->send().  This opens a session with my SMTP server and then sends this single email and promptly closes the SMTP session.  I then repeat this process 500 times.  Not very efficient.

To be more efficient I would like to open an SMTP session, and then $mail->send() each of the 500 emails, and then close the SMTP session.  Does PHPMailer support this???

Cheers
Murray

PS I have been using SwiftMailer but this doesn't support this type of mass emailing.

**Comment by Synchro** [comment]
[Yes](https://github.com/PHPMailer/PHPMailer/wiki/Sending-to-lists), and see [the mailing list example](https://github.com/PHPMailer/PHPMailer/blob/master/examples/mailing_list.phps).

**Reply by murraycollingwood** [reply]
Excellent - exactly what I was after.
